lib/bloc_record/selection.rb

```diff
@@ -33,6 +38,65 @@ def find_by(attribute, value)
     init_object_from_row(row)
   end
 
+  def find_each(options = {}, &block)
+    if options.nil?
+      items = connection.execute <<-SQL
+        SELECT #{columns.join ","} FROM #{table}
+      SQL
+    else
+      items = connection.execute <<-SQL
+        SELECT #{columns.join ","} FROM #{table}
+        LIMIT #{options.size} OFFSET #{options.start};
+      SQL
+    end
+
+    items.each do |item|
+      yield init_object_from_row(item)
+    end
+  end
+
+  def find_in_batches(options = {}, &block)
```

Comment:
Similar to above, except `options` should not be optional. So all bu the first case above should be handled (again, the last one is up to you).